src/www/services_dhcpv6.php

```diff
@@ -34,24 +34,120 @@
+  /* calculated "fixed" range */
+  list ($ifcfgipv6) = interfaces_primary_address6($if, legacy_interfaces_details());
+  $range = ['from' => '?', 'to' => '?'];
+  if (is_ipaddrv6($ifcfgipv6)) {
+      $ifcfgipv6 = Net_IPv6::getNetmask($ifcfgipv6, 64);
+      $ifcfgipv6arr = explode(':', $ifcfgipv6);
+      $ifcfgipv6arr[7] = '1000';
+      $range['from'] = Net_IPv6::compress(implode(':', $ifcfgipv6arr));
+      $ifcfgipv6arr[7] = '2000';
+      $range['to'] = Net_IPv6::compress(implode(':', $ifcfgipv6arr));
+  }
+
+  if (!empty($config['dhcpdv6']) && !empty($config['dhcpdv6']['lan']) && isset ($config['dhcpdv6']['lan']['enable']) && $config['dhcpdv6']['lan']['enable'] == '-1') {
+      /* disabled */
+      $enable_input = '';
+  }
+
+  $range_tr = <<<EOD
+    <tr>
+      <td><i class="fa fa-info-circle text-muted"></i> $range_label</td>
+      <td>{$range['from']} - {$range['to']}</td>
+    </tr>
+  EOD;
+
+
+  echo <<<EOD
+    <section class="page-content-main">
+      <div class="container-fluid">
+        <div class="row">
+          <section class="col-xs-12">
+            <div class="tab-content content-box col-xs-12">
+              <form method="post" name="iform" id="iform">
+                <div class="table-responsive">
+                  <table class="table table-striped opnsense_standard_table_form">
+                    <tr>
+                      <td style="width:22%"></td>
+                      <td style="width:78%; text-align:right">
+                        <div style='height: 15px;'></div>
+                      </td>
+                    </tr>
+                    <tr>
+                      <td><i class="fa fa-info-circle text-muted"></i> $enable_label</td>
+                      <td>
+                        <input name="enable" type="checkbox" value="yes" $enable_input/>
+                        <strong>$enable_descr</strong>
+                      </td>
+                      $range_tr
+                    </tr>
+                    <tr>
+                      <td>&nbsp;</td>
+                      <td>
+                        <input name="if" type="hidden" value="$if" />
+                        <input name="submit" type="submit" class="formbtn btn btn-primary" value="$save_btn_text"/>
+                      </td>
+                    </tr>
+                  </table>
+                </div>
+              </form>
+            </div>
+          </section>
+        </div>
+      </div>
+    </section>
+  EOD;
+  include("foot.inc");
+}
+
+function process_track6_form($if)
+{
+    $dhcpdv6cfg = &config_read_array('dhcpdv6');
+    $this_server = [];
+    if (isset($dhcpdv6cfg[$if]) && isset($dhcpdv6cfg[$if]['ramode'])) {
+        /* keep ramode for router advertisements so we can use this field to disable the service when in tracking mode */
+        $this_server['ramode'] = $dhcpdv6cfg[$if]['ramode'];
+    }
+    if (empty($_POST['enable'])) {
+        $this_server['enable'] = '-1';
```

Comment:
the downside of this is when the interface is switched to static the value will be '' or '0' again and it re-enabled the auto-mode which potentially leads to more code and all the 'enable' use flags need to be checked/adjusted too empty() doesn't work anymore. Maybe you have done it, i just want to mention it here.

Reply:
I don't think the interfaces page mangles with `dhcpv6` section, the only reference I could find was this one :

https://github.com/opnsense/core/blob/39d5ff1a26ae306abc83c9ae1c70c8c4c07ce997/src/www/interfaces.php#L624-L626


which is merely a validation (and might need a change, but haven't looked at that yet)

Reply:
yeah stuff like that, just to keep in mind